tests/assembly/riscv32e-registers.rs

```diff
@@ -0,0 +1,108 @@
+// Test that loads into registers x16..=x31 are never generated for riscv32e
+// targets.
+//
+//@ revisions:     x1  x2  x3  x4  x5  x6  x7  x8  x9  x10 x11 x12 x13 x14 x15
+//@ revisions: x16 x17 x18 x19 x20 x21 x22 x23 x24 x25 x26 x27 x28 x29 x30 x31
+//@ assembly-output: emit-asm
+//@ compile-flags: --target riscv32e-unknown-none-elf
+//@ filecheck-flags: --check-prefix riscv32e --check-prefix riscv32em --check-prefix riscv32emc
+//@ needs-llvm-components: riscv
+//
+//@ [x16] should-fail
+//@ [x17] should-fail
+//@ [x18] should-fail
+//@ [x19] should-fail
+//@ [x20] should-fail
+//@ [x21] should-fail
+//@ [x22] should-fail
+//@ [x23] should-fail
+//@ [x24] should-fail
+//@ [x25] should-fail
+//@ [x26] should-fail
+//@ [x27] should-fail
+//@ [x28] should-fail
+//@ [x29] should-fail
+//@ [x30] should-fail
+//@ [x31] should-fail
+
+#![feature(no_core, lang_items, rustc_attrs)]
+#![crate_type = "rlib"]
+#![no_core]
+
+#[rustc_builtin_macro]
+macro_rules! asm {
+    () => {};
+}
+
+#[lang = "sized"]
+trait Sized {}
+
+// Check that a load instruction is generated for each register unless explicitly failed
+// CHECK: #APP
+// CHECK: li {{[a-z]+}}{{[0-9]*}}, 0
+// CHECK: #NO_APP
```

Comment:
I do have a question though: Where are the `APP` and `NO_APP` variables getting set?

Reply:
Huh, I thought that was a FileCheck intrinsic. I'll re-verify what I've done.

Reply:
It might be...? If so, it's not in the FileCheck docs...

Reply:
Old test removed, above is obsolete.